In [1]:
import numpy as np


In [14]:
def data_sample():
    mu = 6
    W = np.random.normal(4, 2) # mean is 4, variance is sqrt(4)
    return mu + W

samples = [data_sample() for i in range(50)]


In [34]:
EW = 4
VarW = 4
n = 50
Ybar = np.mean(np.asarray(samples))
lower_bound = Ybar - EW - 1.96 * np.sqrt(VarW / n)

upper_bound = Ybar - EW + 1.96 * np.sqrt(VarW / n)

print('Lower bound: ', lower_bound)
print('Upper bound: ', upper_bound)

Lower bound:  5.265628283549747
Upper bound:  6.374371716450254


c) The true value does lie within this interval

In [35]:
n_trials = 10000
samples_per_trial = 100
results = [] # this will be a list of booleans
within_interval = 0
mu = 6
for i in range(n_trials):
    samples = [data_sample() for i in range(samples_per_trial)]
    EW = 4
    VarW = 4
    Ybar = np.mean(np.asarray(samples))
    lower_bound = Ybar - EW - 1.96 * np.sqrt(VarW / samples_per_trial)
    upper_bound = Ybar - EW + 1.96 * np.sqrt(VarW / samples_per_trial)
    
    if ((mu >= lower_bound) and (mu <= upper_bound)):
        within_interval += 1

p_within = within_interval / n_trials
print(p_within)
    

0.9527


The percent of the time the value falls within the bounds is 95.27%. This matches the 95% confidence interval. 

In [26]:
def poiss_data_sample():
    mu = 6
    W = np.random.poisson(lam=4) # mean is 4, variance is sqrt(4)
    return mu + W
poiss_samples = [poiss_data_sample() for _ in range(50)]



In [27]:
# mean and variance are same in possion RV

EW = 4
VarW = 4
n = 50
Ybar = np.mean(np.asarray(poiss_samples))
lower_bound = Ybar - EW - 1.96 * np.sqrt(VarW / n)
upper_bound = Ybar - EW + 1.96 * np.sqrt(VarW / n)

print('Lower bound: ', lower_bound)
print('Upper bound: ', upper_bound)

Lower bound:  5.605628283549747
Upper bound:  6.714371716450254


c) Yes, the true value is within the bounds 

In [36]:
n_trials = 10000
samples_per_trial = 50
results = [] # this will be a list of booleans
within_interval = 0
mu = 6
for i in range(n_trials):
    samples = [poiss_data_sample() for i in range(samples_per_trial)]
    EW = 4
    VarW = 4
    n = 50
    Ybar = np.mean(np.asarray(poiss_samples))
    lower_bound = Ybar - EW - 1.96 * np.sqrt(VarW / n)
    upper_bound = Ybar - EW + 1.96 * np.sqrt(VarW / n)
    
    if ((mu >= lower_bound) and (mu <= upper_bound)):
        within_interval += 1

p_within = within_interval / n_trials
print(p_within)

1.0


The samples fall within the bound 100% of the time. 